java/openjdk-java/Portfile

```
@@ -0,0 +1,302 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+
+PortSystem          1.0
+
+name                openjdk-java
+# Latest Long Term Support (LTS) major version
+version             17
+revision            0
+description         Replaced by openjdk${version}u
+categories          java devel
+platforms           darwin
+supported_archs     noarch
+license             GPL-2+
+maintainers         nomaintainer
+universal_variant   no
+
+set long_description_openjdk-ga \
+    "Production-ready open-source builds of the Openjdk, \
+    Open Source implementation of the Java Platform, Standard Edition, and related projects."
+
+set long_description_adoption-jdk \
+    "The adoption group of OpenJDK was formed out of the global \
+    Java User Group (JUG) Adopt OpenJDK programme \
+    A majority of the original programme is expected \
+    to be folded up into this adoption group over time. \
+    Although AdoptOpenJDK is not an official OpenJDK project, \
+    most OpenJDK's Adoption Group members are involved in the new shared build farm for OpenJDK at adoptopenjdk.net. \
+    This build farm, infrastructure and test scripts as code will allow OpenJDK developers \
+    to build and test OpenJDK on as many versions, variants and platforms as possible."
+
+
+set obsoleted_ports {
+    openjdk-java
+}
+
+if {${subport} in ${obsoleted_ports}} {
+    PortGroup    obsolete 1.0
+}
+
+if {${subport} eq "openjdk-java"} {
+    # The openjdk port is not installable, but recommends users to install the latest Long Term Support (LTS) subport
+    replaced_by  openjdk${version}u
+}
+
+# For openjdk repository :  Use 'https://git.openjdk.java.net/(git repository name from https://github.com/openjdk)' format for repository
+# For openjdk*u subports : Use the openjdk[Major version]-boot for boot jdk and Use the openjdk[Major versionu]u for openjdk
+# openjdk8-adoption-boot is named as boot jdk subport for openjdk8u as it uses adoptopenjdk which is though an unofficial openjdk but it is maintained by Adoption group at Openjdk (https://wiki.openjdk.java.net/display/Adoption/Main)
+# For openjdk 9 or later : Use boot jdk as mentioned in Openjdk's installation guide (https://openjdk.java.net/install/)
+
+subport openjdk8-adoption-boot {
+    # https://wiki.openjdk.java.net/display/Adoption/Main
+    # https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/jdk8u-2018-11-07-15-21/
+    version             8u192-b12
+    description         Openjdk 8 AdoptOpenJDK
+    long_description    ${long_description_adoption-jdk}
```

For `description` fields, if using another variable, please use Tcl's argument expansion (`{*}`):
```suggestion
    long_description    {*}${long_description_adoption-jdk}
```

---

@herbygillot I'm not very familiar with Tcl, and I wonder why this is necessary. It seems this pattern was copied from what I do in the existing `openjdk` portfile.

---

`{*}` is required for `description` field, if using another variable, as it promotes the text in the variable to individual arguments.

---

That's correct @usersxx.  @breun `{*}` is Tcl's argument expansion operator.  It takes a group containing multiple items and expands the items out to be individual values.

```
% p {a b c}
Number of arguments p sees: 1
% p {*}{a b c}
Number of arguments p sees: 3
```

...from https://wiki.tcl-lang.org/page/%7B%2A%7D